lib-multisrc/manhwaz/src/eu/kanade/tachiyomi/multisrc/manhwaz/ManhwaZ.kt

```diff
@@ -0,0 +1,275 @@
+        genre = document.select("div.genres-content a[rel=tag]").joinToString { it.text() }
+        status = when {
+            ongoingStatusList.contains(statusText) -> SManga.ONGOING
+            completedStatusList.contains(statusText) -> SManga.COMPLETED
+            else -> SManga.UNKNOWN
+        }
+        thumbnail_url = document.selectFirst("div.summary_image img")?.imgAttr()
+    }
+
+    override fun chapterListSelector() = "li.wp-manga-chapter"
+
+    override fun chapterFromElement(element: Element) = SChapter.create().apply {
+        element.selectFirst("a")!!.let {
+            setUrlWithoutDomain(it.attr("href"))
+            name = it.text()
+        }
+
+        element.selectFirst("span.chapter-release-date")?.text()?.let {
+            date_upload = parseRelativeDate(it)
+        }
+    }
+
+    override fun pageListParse(document: Document) =
+        document.select("div.page-break img").mapIndexed { i, it ->
+            Page(i, imageUrl = it.imgAttr())
+        }
+
+    override fun imageUrlParse(document: Document) = throw UnsupportedOperationException()
+
+    override fun getFilterList(): FilterList {
+        fetchGenreList()
+
+        val filters = buildList {
+            add(Filter.Header(intl["filter_ignored_warning"]))
+            add(Filter.Header(intl.format("cannot_use_order_by_warning", intl["genre_all"], intl["genre_completed"])))
+
+            if (fetchGenreStatus == FetchGenreStatus.NOT_FETCHED && fetchGenreAttempts >= 3) {
+                add(Filter.Header(intl["genre_fetch_failed"]))
+            } else if (fetchGenreStatus != FetchGenreStatus.FETCHED) {
+                add(Filter.Header(intl["genre_missing_warning"]))
+            }
+
+            add(Filter.Separator())
+            if (genres.isNotEmpty()) {
+                add(GenreFilter(intl, genres))
+            }
+            add(OrderByFilter(intl))
+        }
+
+        return FilterList(filters)
+    }
+
+    private class GenreFilter(
+        intl: Intl,
+        genres: List<SelectOption>,
+    ) : SelectFilter(intl["genre_filter_title"], genres)
+
+    private class OrderByFilter(intl: Intl) : SelectFilter(
+        intl["order_by_filter_title"],
+        listOf(
+            SelectOption(intl["order_by_latest"], "latest"),
+            SelectOption(intl["order_by_rating"], "rating"),
+            SelectOption(intl["order_by_most_views"], "views"),
+            SelectOption(intl["order_by_new"], "new"),
+        ),
+    )
+
+    private var genres = emptyList<SelectOption>()
+    private var fetchGenreStatus = FetchGenreStatus.NOT_FETCHED
+    private var fetchGenreAttempts = 0
+
+    @OptIn(DelicateCoroutinesApi::class)
+    private fun fetchGenreList() {
+        if (fetchGenreStatus != FetchGenreStatus.NOT_FETCHED || fetchGenreAttempts >= 3) {
+            return
+        }
+
+        fetchGenreStatus = FetchGenreStatus.FETCHING
+        fetchGenreAttempts++
+
+        GlobalScope.launch(Dispatchers.IO) {
```

Comment:
CoroutineScope?

Reply:
Didn't we talk about not using CoroutineScope before? https://github.com/keiyoushi/extensions-source/pull/1292#discussion_r1493317304

Reply:
Ah nevermind, you fixed it to use a CoroutineScope already.